src/coreclr/src/tools/r2rdump/R2RDiff.cs

```diff
@@ -209,11 +287,16 @@ private void ShowDiff(Dictionary<string, int> leftObjects, Dictionary<string, in
         /// </summary>
         /// <param name="reader">R2R image to scan</param>
         /// <returns></returns>
-        private Dictionary<string, int> GetR2RMethodMap(ReadyToRunReader reader)
+        private Dictionary<string, int> GetR2RMethodMap(ReadyToRunReader reader, ReadyToRunSection section)
         {
             Dictionary<string, int> methodMap = new Dictionary<string, int>();
 
-            foreach (ReadyToRunMethod method in reader.Methods)
+            if (!reader.Methods.TryGetValue(section, out List<ReadyToRunMethod> sectionMethods))
+            {
+                sectionMethods = new List<ReadyToRunMethod>();
```

Comment:
It seems we could just flip the condition and put the foreach inside of the if block instead of creating an empty list that we throw away right after we try to iterate over it.

Reply:
Hmm, stupid me, no idea why I didn't do it like that in the first place, perhaps some copy & paste from elsewhere in the file. Thanks for pointing that out!